hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/snapshot/OMSnapshotPurgeRequest.java

```diff
@@ -145,7 +145,6 @@ private void updateSnapshotInfoAndCache(SnapshotInfo snapInfo,
       // Update table cache first
       omMetadataManager.getSnapshotInfoTable().addCacheEntry(new CacheKey<>(snapInfo.getTableKey()),
           CacheValue.get(trxnLogIndex, snapInfo));
-      updatedSnapInfos.put(snapInfo.getTableKey(), snapInfo);
```

Comment:
We need to add this as well. Because `updateSnapshotInfoAndCache` is being called by 
https://github.com/apache/ozone/blob/ab343da40d1dedb7a71a47946ce4eda79015a0dd/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/snapshot/OMSnapshotPurgeRequest.java#L108-L112

This `nextSnapshot` should also be added to `updatedSnapshotInfos` because we are not using `getUpdatedSnapshotInfo` when getting this snapshot so it won't be added to the cache.

Reply:
Good catch. Thanks for the catch. Updated.

Reply:
I'll also create a follow-up task to add tests to cover this.